pallets/parachain-staking/src/lib.rs

```diff
@@ -432,29 +441,26 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(now: T::BlockNumber) -> frame_support::weights::Weight {
+			let mut post_weight = T::DbWeight::get().reads(1);
 			let mut round = <Round<T>>::get();
+
+			// check for round update
 			if round.should_update(now) {
 				// mutate round
 				round.update(now);
 
 				// start next round
 				<Round<T>>::put(round);
 
-				// TODO: Check whether we want to remove the storage read if we reduce the data
-				// provided by Event::NewRound
-				let TotalStake {
-					collators: collator_staked,
-					delegators: delegator_staked,
-				} = <Total<T>>::get();
-				Self::deposit_event(Event::NewRound(
-					round.first,
-					round.current,
-					collator_staked,
-					delegator_staked,
-				));
-				T::DbWeight::get().reads_writes(2, 1)
+				Self::deposit_event(Event::NewRound(round.first, round.current));
+				post_weight = post_weight.saturating_add(T::DbWeight::get().writes(1));
+			}
+			// check for InflationInfo update
+			if now > T::BlocksPerYear::get() {
```

Comment:
We can maybe assign the result of the if-else expression to `post_weight` and return that?

Reply:
Should be fixed in 309251c.